lib/parser/parser.ts

```diff
@@ -51,60 +51,71 @@ export class Parser extends ParserBase {
   }
   parse(): OFile {
 
-    let disabledRangeStart = undefined;
-    const ignoreRegex: RegExp[] = [];
+    const disabledRangeStart = new Map<string | undefined, number>();
     for (const [lineNumber, line] of this.originalText.split('\n').entries()) {
       const match = /(--\s*vhdl-linter)(.*)/.exec(line); // vhdl-linter-disable-next-line //vhdl-linter-disable-this-line
 
       if (match) {
         let innerMatch: RegExpMatchArray | null;
         const nextLineRange = new OIRange(this.file, new OI(this.file, lineNumber + 1, 0), new OI(this.file, lineNumber + 1, this.originalText.split('\n')[lineNumber + 1].length - 1));
-        if ((innerMatch = match[2].match('-disable-this-line')) !== null) {
-          this.file.magicComments.push(new OMagicCommentDisable(this.file, MagicCommentType.Disable, new OIRange(this.file, new OI(this.file, lineNumber, 0), new OI(this.file, lineNumber, line.length - 1))));
-        } else if ((innerMatch = match[2].match('-disable-next-line')) !== null) {// TODO: next nonempty line
-          this.file.magicComments.push(new OMagicCommentDisable(this.file, MagicCommentType.Disable, nextLineRange));
-        } else if ((innerMatch = match[2].match('-disable-region')) !== null) {
-          if (disabledRangeStart === undefined) {
-            disabledRangeStart = lineNumber;
+        if ((innerMatch = match[2].match(/-disable(?:-this)?-line(?:\s|$)(.+)?/i)) !== null) {
+          for (const rule of innerMatch[1]?.split(' ') ?? [undefined]) {
+            this.file.magicComments.push(new OMagicCommentDisable(
+              this.file,
+              MagicCommentType.Disable,
+              new OIRange(this.file, new OI(this.file, lineNumber, 0), new OI(this.file, lineNumber, line.length - 1)),
```

Comment:
```suggestion
              new OIRange(this.file, new OI(this.file, lineNumber, 0), new OI(this.file, lineNumber, line.length)),
```
When messages are created with a range of `getRangeToEndLine` it is not correctly disabled.
An additional test for this:
```vhdl
library ieee;
use ieee.std_logic_1164.all;
use ieee.numeric_std.all;

entity test_procedure_call is
end entity;

architecture rtl of test_procedure_call is
begin
  process
    procedure p(a : in std_ulogic; b: out std_ulogic) is
    begin
      b <= a;
    end procedure;
  begin
    p(a => '1', 32); -- vhdl-linter-disable-line instantiation
  end process;

end architecture;
```